src/aui/auibook.cpp

```diff
@@ -3446,6 +3446,127 @@ bool wxAuiNotebook::InsertPage(size_t index, wxWindow *page,
     }
 }
 
+class wxAuiLayoutObject
```

Comment:
I'd very much appreciate a comment explaining what's going on here. AFAICS the main purpose of this class is to be able to call `MergeLayout()` on it, but I don't really know _why_ do we need to do it.

Reply:
I added comments to all critical parts of the algorithm.